Error building SMACK from source on older versions of Ubuntu >18.04

I'm trying to build SMACK from source on Ubuntu 16.04.7 LTS using the `bin/build.sh` script, but I'm getting the following error:
```console
-- Build files have been written to: /home/ubuntu/smack/build
[12/63] Building CXX object CMakeFiles/sma...kTranslator.dir/lib/smack/DSAWrapper.cpp.o
FAILED: /usr/bin/c++   -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I../include -I../sea-dsa/include -I/usr/lib/llvm-11/include -std=c++14   -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -fno-exceptions -fno-rtti -O3 -fPIC -fvisibility-inlines-hidden -Werror=date-time -w -fdiagnostics-color -g -MMD -MT CMakeFiles/smackTranslator.dir/lib/smack/DSAWrapper.cpp.o -MF CMakeFiles/smackTranslator.dir/lib/smack/DSAWrapper.cpp.o.d -o CMakeFiles/smackTranslator.dir/lib/smack/DSAWrapper.cpp.o -c ../lib/smack/DSAWrapper.cpp
In file included from ../sea-dsa/include/seadsa/CallSite.hh:12:0,
                 from ../sea-dsa/include/seadsa/BottomUp.hh:9,
                 from ../sea-dsa/include/seadsa/Global.hh:9,
                 from ../sea-dsa/include/seadsa/DsaAnalysis.hh:9,
                 from ../include/smack/DSAWrapper.h:13,
                 from ../lib/smack/DSAWrapper.cpp:7:
../sea-dsa/include/seadsa/Graph.hh: In member function ‘constexpr std::tuple<unsigned int, seadsa::FieldType> seadsa::Field::asTuple() const’:
../sea-dsa/include/seadsa/Graph.hh:301:29: error: converting to ‘std::tuple<unsigned int, seadsa::FieldType>’ from initializer list would use explicit constructor ‘constexpr std::tuple<_T1, _T2>::tuple(const _T1&, const _T2&) [with _T1 = unsigned int; _T2 = seadsa::FieldType]’
     return {m_offset, m_type};
                             ^
[12/63] Building CXX object CMakeFiles/smackTranslator.dir/lib/smack/Regions.cpp.o
FAILED: /usr/bin/c++   -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I../include -I../sea-dsa/include -I/usr/lib/llvm-11/include -std=c++14   -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -fno-exceptions -fno-rtti -O3 -fPIC -fvisibility-inlines-hidden -Werror=date-time -w -fdiagnostics-color -g -MMD -MT CMakeFiles/smackTranslator.dir/lib/smack/Regions.cpp.o -MF CMakeFiles/smackTranslator.dir/lib/smack/Regions.cpp.o.d -o CMakeFiles/smackTranslator.dir/lib/smack/Regions.cpp.o -c ../lib/smack/Regions.cpp
In file included from ../include/smack/Regions.h:7:0,
                 from ../lib/smack/Regions.cpp:4:
../sea-dsa/include/seadsa/Graph.hh: In member function ‘constexpr std::tuple<unsigned int, seadsa::FieldType> seadsa::Field::asTuple() const’:
../sea-dsa/include/seadsa/Graph.hh:301:29: error: converting to ‘std::tuple<unsigned int, seadsa::FieldType>’ from initializer list would use explicit constructor ‘constexpr std::tuple<_T1, _T2>::tuple(const _T1&, const _T2&) [with _T1 = unsigned int; _T2 = seadsa::FieldType]’
     return {m_offset, m_type};
                             ^
[12/63] Building CXX object CMakeFiles/sma...r.dir/lib/smack/SmackModuleGenerator.cpp.o
FAILED: /usr/bin/c++   -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I../include -I../sea-dsa/include -I/usr/lib/llvm-11/include -std=c++14   -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -fno-exceptions -fno-rtti -O3 -fPIC -fvisibility-inlines-hidden -Werror=date-time -w -fdiagnostics-color -g -MMD -MT CMakeFiles/smackTranslator.dir/lib/smack/SmackModuleGenerator.cpp.o -MF CMakeFiles/smackTranslator.dir/lib/smack/SmackModuleGenerator.cpp.o.d -o CMakeFiles/smackTranslator.dir/lib/smack/SmackModuleGenerator.cpp.o -c ../lib/smack/SmackModuleGenerator.cpp
In file included from ../include/smack/Regions.h:7:0,
                 from ../lib/smack/SmackModuleGenerator.cpp:10:
../sea-dsa/include/seadsa/Graph.hh: In member function ‘constexpr std::tuple<unsigned int, seadsa::FieldType> seadsa::Field::asTuple() const’:
../sea-dsa/include/seadsa/Graph.hh:301:29: error: converting to ‘std::tuple<unsigned int, seadsa::FieldType>’ from initializer list would use explicit constructor ‘constexpr std::tuple<_T1, _T2>::tuple(const _T1&, const _T2&) [with _T1 = unsigned int; _T2 = seadsa::FieldType]’
     return {m_offset, m_type};
                             ^
[12/63] Building CXX object CMakeFiles/sma...ator.dir/lib/smack/CodifyStaticInits.cpp.o
FAILED: /usr/bin/c++   -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I../include -I../sea-dsa/include -I/usr/lib/llvm-11/include -std=c++14   -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -fno-exceptions -fno-rtti -O3 -fPIC -fvisibility-inlines-hidden -Werror=date-time -w -fdiagnostics-color -g -MMD -MT CMakeFiles/smackTranslator.dir/lib/smack/CodifyStaticInits.cpp.o -MF CMakeFiles/smackTranslator.dir/lib/smack/CodifyStaticInits.cpp.o.d -o CMakeFiles/smackTranslator.dir/lib/smack/CodifyStaticInits.cpp.o -c ../lib/smack/CodifyStaticInits.cpp
In file included from ../sea-dsa/include/seadsa/CallSite.hh:12:0,
                 from ../sea-dsa/include/seadsa/BottomUp.hh:9,
                 from ../sea-dsa/include/seadsa/Global.hh:9,
                 from ../sea-dsa/include/seadsa/DsaAnalysis.hh:9,
                 from ../include/smack/DSAWrapper.h:13,
                 from ../lib/smack/CodifyStaticInits.cpp:8:
../sea-dsa/include/seadsa/Graph.hh: In member function ‘constexpr std::tuple<unsigned int, seadsa::FieldType> seadsa::Field::asTuple() const’:
../sea-dsa/include/seadsa/Graph.hh:301:29: error: converting to ‘std::tuple<unsigned int, seadsa::FieldType>’ from initializer list would use explicit constructor ‘constexpr std::tuple<_T1, _T2>::tuple(const _T1&, const _T2&) [with _T1 = unsigned int; _T2 = seadsa::FieldType]’
     return {m_offset, m_type};
                             ^
[12/63] Building CXX object CMakeFiles/sma...tor.dir/lib/smack/SmackInstGenerator.cpp.o
ninja: build stopped: subcommand failed.
```

I tried to use `-std=c++14` flag or to use `clang`, but neither worked.

I also try these [instructions](https://github.com/smackers/smack/blob/master/docs/installation.md#super-quick-setup-virtual-smack) to get `vsmack`, but the link seems broken. Any advice?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -200,6 +200,7 @@ install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/share/smack
   USE_SOURCE_PERMISSIONS
   FILES_MATCHING PATTERN "*.py" PATTERN "*.h" PATTERN "*.c" PATTERN "Makefile"
     PATTERN "*.rs" PATTERN "*.f90" PATTERN "*.di" PATTERN "*.toml"
+    PATTERN "*.cpp" PATTERN "cassert"
 )
 
 install(FILES

--- a/share/smack/include/cassert
+++ b/share/smack/include/cassert
@@ -1,0 +1,4 @@
+//
+// This file is distributed under the MIT License. See LICENSE for details.
+//
+#include <assert.h>

